File: v131016/TeeNew/Title_ClickEvent.pas
unit Title_ClickEvent;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  {$IFDEF CLR}
  Types,
  {$ENDIF}
  Base, TeEngine, Series, TeeProcs, Chart;

type
  TTitleClickEvent = class(TBaseForm)
    Series1: TPieSeries;
    procedure FormCreate(Sender: TObject);
    procedure Chart1ClickTitle(Sender: TCustomChart; ATitle: TChartTitle;
      Button: TMouseButton; Shift: TShiftState; X, Y: Integer);
    procedure Chart1MouseMove(Sender: TObject; Shift: TShiftState; X,
      Y: Integer);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TTitleClickEvent.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(6);
end;

procedure TTitleClickEvent.Chart1ClickTitle(Sender: TCustomChart;
  ATitle: TChartTitle; Button: TMouseButton; Shift: TShiftState; X,
  Y: Integer);
begin
  Panel1.Caption:=ATitle.Text.Text;
end;

procedure TTitleClickEvent.Chart1MouseMove(Sender: TObject;
  Shift: TShiftState; X, Y: Integer);
var P : TChartClickedPart;
begin
  Chart1.CalcClickedPart(Point(x,y),P);
  Case P.Part of
    cpTitle,cpFoot,
    cpSubTitle,cpSubFoot: Chart1.Cursor:=crHandPoint;
  else
    Chart1.Cursor:=crDefault;
  end;
  Chart1.OriginalCursor:=Chart1.Cursor;
end;

initialization
  RegisterClass(TTitleClickEvent);
end.

File: TeeNew/Builder2006/Title_ClickEvent.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Title_ClickEvent.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma resource "*.dfm"
TTitleClickEvent *TitleClickEvent;
//---------------------------------------------------------------------------
__fastcall TTitleClickEvent::TTitleClickEvent(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TTitleClickEvent::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(6);
}
//---------------------------------------------------------------------------

void __fastcall TTitleClickEvent::Chart1ClickTitle(TCustomChart *Sender,
      TChartTitle *ATitle, TMouseButton Button, TShiftState Shift, int X,
      int Y)
{
  Panel1->Caption = ATitle->Text->Text;
}
//---------------------------------------------------------------------------

void __fastcall TTitleClickEvent::Chart1MouseMove(TObject *Sender,
      TShiftState Shift, int X, int Y)
{
  TChartClickedPart P;
  Chart1->CalcClickedPart(Point(X,Y),P);

  switch (P.Part) {
    case cpTitle    :
    case cpFoot     :
    case cpSubTitle :
    case cpSubFoot  : Chart1->Cursor = crHandPoint;
         break;
    default : Chart1->Cursor = crDefault;
  }

  Chart1->OriginalCursor = Chart1->Cursor;
}
//---------------------------------------------------------------------------

